lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -188,7 +188,9 @@ def parse_relationship(assoc_name, assoc_data, options)
             end
 
           polymorphic = (options[:polymorphic] || []).include?(assoc_name.to_sym)
-          hash["#{prefix_key}_type".to_sym] = assoc_data[:type] if polymorphic
+          if polymorphic
+            hash["#{prefix_key}_type".to_sym] = assoc_data.present? ? assoc_data[:type] : nil
```

Comment:
assoc_data is nil on non-master rails, and empty string on master rails. but yeah, the idea is that if the assoc_data isn't there, it would still be nice to get the _id and _type attributes in the resulting hash, so at least you know your front end is aware of the relationship (but hasn't set it)